qemu/tests/vioinput_keyboard.py

```diff
@@ -78,6 +78,7 @@ def key_check(key):
 
     error_context.context("Start event listener in guest", logging.info)
     listener = input_event_proxy.EventListener(vm)
+    time.sleep(30)
```

Comment:
Need to know what really happened at that time and give a reasonable clarification, otherwise, I'm wondering it should be a product bug.

And also, even if we need this check, it should be done within the listener.

Reply:
Hi xuhan, I cannot reproduce this issue by manually although with new installed image, seems my operation speed slower than automation reproduced. assume my operation speed is instant with customer operation speed, it's should not belong a product issue. Indeed, I also tried to know what really happened, spent much time to debug, but I could not get the deeper reason between listener starting and send key, I tried tests through add waiting time from 5secs to 30secs at this time, found wait >=30secs can resolve this issue, so can I fix it as a workaround for this situation or could you help to check it?
Thanks~

Reply:
what's the error you got?

Reply:
This one: avocado.core.exceptions.TestFail: Reveived key events [] were not enough

and details as:
04:22:18 ERROR| Reproduced traceback from: /usr/local/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/core/test.py:860
04:22:18 ERROR| Traceback (most recent call last):
04:22:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/avocado_vt/test.py", line 425, in _runTest
04:22:18 ERROR|     run_func(self, params, env)
04:22:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
04:22:18 ERROR|     return fn(*args, **kwargs)
04:22:18 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/vioinput_keyboard.py", line 125, in run
04:22:18 ERROR|     key_tap_test(test, params, vm)
04:22:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
04:22:18 ERROR|     return fn(*args, **kwargs)
04:22:18 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/vioinput_keyboard.py", line 89, in key_tap_test
04:22:18 ERROR|     key_check(key)
04:22:18 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/vioinput_keyboard.py", line 53, in key_check
04:22:18 ERROR|     test.fail("Reveived key events %s were not enough" % key_event_lst)
04:22:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/core/test.py", line 1050, in fail
04:22:18 ERROR|     raise exceptions.TestFail(message)
04:22:18 ERROR| avocado.core.exceptions.TestFail: Reveived key events [] were not enough

Reply:
And this is log file: https://bugzilla.redhat.com/attachment.cgi?id=1668616

Reply:
From the content of screen dump we can see, the guest os was having device driver installed at that time, and it seems to be the root cause. If so you have to check whether virtio input devices were ready to use before doing the key test, rather than waiting for a few seconds.

Reply:
It's wired that I do check vioinput driver is running before key test, and in debug log also can found the check command 
09:48:37 INFO | Context: Check vioinput driver is running
09:48:37 INFO | Context: Check vioinput driver is running --> Check vioinput driver state.
09:48:37 DEBUG| Sending command: wmic sysdriver where PathName="C:\\Windows\\System32\\drivers\\vioinput.sys" get State /value
And shown running in session log:
2020-03-18 09:48:39: State=Running

May be only this check is not enough, I'll deeply find any other check needed here, thanks a lot~